scripts/annotate_study.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+"""
+Create a map-annotation on a top-level object based on an IDR study file.
+All fields are treated as optional
+Example:
+
+    scripts/annotate_study.py \
+        idr-metadata/idr0001-graml-sysgro/idr0001-study.txt Screen:3
+"""
+
+import argparse
+import csv
+
+import omero.clients
+from omero.cli import CLI, ProxyStringType
+from omero.model import MapAnnotationI, NamedValue
+from omero.rtypes import rstring
+
+
+def create_map_annotation(targets, rowkvs, ns):
+    ma = MapAnnotationI()
+    ma.setNs(rstring(ns))
+    mv = []
+    for k, vs in rowkvs:
+        if not isinstance(vs, (tuple, list)):
+            vs = [vs]
+        mv.extend(NamedValue(k, str(v)) for v in vs)
+
+    if not mv:
+        raise Exception('Empty MapValue')
+
+    ma.setMapValue(mv)
+
+    links = []
+    for target in targets:
+        otype = target.ice_staticId().split('::')[-1]
+        link = getattr(omero.model, '%sAnnotationLinkI' % otype)()
+        link.setParent(target)
+        link.setChild(ma)
+        links.append(link)
+    return links
+
+
+# Mapping of
+# - study keys (from the study file)
+# - displayed names
+# - displayed values (default value)
+study_keys = [
+    ('Study PubMed ID', 'PubMed ID'),
+    ('Study PubMed ID', 'PubMed ID URL',
+     'http://www.ncbi.nlm.nih.gov/pubmed/{{ value }}'),
+    ('Study DOI', 'Publication DOI'),
+    ('Study DOI', 'Publication DOI URL', 'http://dx.doi.org/{{ value }}'),
+    ('Study Publication Title', 'Title'),
+    ('Study Author List', 'Authors'),
+    ('Study External URL', 'External URL'),
+]
+
+ns = 'openmicroscopy.org/idr/study'
```

Comment:
Does this mean all existing study would need to be reannotated with the namespace?

Reply:
@pwalczysko mentioned this Friday. The easiest option is to change this to the current client annotation namespace. However I think the ideal option is to use a dedicated namespace since in principle this information should be discoverable. Thoughts?

Reply:
>to use a dedicated namespace since in principle this information should be discoverable

I am afraid I do not understand. What information should be discoverable ? What advantages does a dedicated namespace give over the current client annotation namespace  which I can and must in some cases edit later ?

Reply:
It mean users and developers can discover all the study annotations through the API by querying for annotations in `openmicroscopy.org/idr/study` or similar. The editable namespace is `openmicroscopy.org/omero/client/mapAnnotation` which is UI centric and doesn't confer any meaning to a user of the IDR.

Reply:
Two comments:

- from the developer perspective, overall I like the idea of the study namespace with the usual caveats about 1- naming and 2- grouping. Two questions: 1- should `idr` be in the namespace as it is in the URL/hostname for the query? 2- we have map annotations for studies but also top-level projects and screens in the case of multiple experiments and screens. Would you propose to use the same namespace or specializing it?

- from the importer/annotator perspective, the main caveat @pwalczysko faced is that the non-client namespace prevents using the UI to perform any modification. Would you see these happening via the script of the CLI?

Reply:
1. `openmicroscopy.org/idr/` is historical, I'm happy with moving to `idr.openmicroscopy.org`, or if you think this namespace is applicable outside the IDR then something else?
2. Ideally all annotations would be scripted since this minimises the risk of human errors creeping in. One of the original goals of the IDR was that in principle each study should be reconstructable entirely automatically.

Reply:
>...study should be reconstructable entirely automatically.

this approach has one significant drawback -> the time is not accounted for, because the pressure here is to deliver some bits before other bits. Thus, in real time, the info is incomplete. Completion is achieved only post-hoc as I understand it.  The script should provision for that  ?

Reply:
Assuming the information is always present in the study file we could add a `delete and replace` flag to this script. First we must decide whether we want to use this script as the basis for annotating all studies, or if we want to take a different approach.